src/utils/parseData.js

```diff
@@ -6,142 +7,150 @@
  * */
 
 export default function parseData(spec, datum) {
-  if (!datum || !spec) return null;
-  let parsedDatum = {};
-
-  // if datum is an array, recursively loop through it
-  if (Array.isArray(datum)) {
-    const parsedData = [];
-    for (let d = 0; d < datum.length; d += 1) {
-      parsedDatum = parseData(spec, datum[d].properties || datum[d]);
-      if (parsedDatum && Object.keys(parsedDatum).length) {
-        parsedData.push(datum[d].properties
-          ? { ...datum[d], properties: parsedDatum }
-          : parsedDatum);
-      }
+    if (!datum || !spec) return null;
+    let parsedDatum = {};
+    // Parse data to number and floats respectively
+    if (Array.isArray(spec)) {
+        spec.forEach((config) => {
+            config.field.forEach(field => {
+                if (config.type === 'number') {
+                    datum[field] = parseInt(datum[field], 10);
+                } else if (config.type === 'float') {
+                    datum[field] = parseFloat(datum[field]);
+                }
+            })
+        });
+        return datum
+    }
```

Comment:
@cKellyDesign No this won't be happening if the spec is not an array. I now realize this will pose a challenge in the event we have our normal data-parse and we still want to parse types. I'm thinking the ultimate solution would be to introduce a new prop on the spec maybe type-parse then have that handling type parsing. please share your thoughts on this when you have a chance

Reply:
@kahummer if it's supposed to happen every time, could we do something like this?
```
(Array.isArray(spec) ? spec : [spec]).forEach(...)
```